backend/msgpack/writer.go

```diff
@@ -209,6 +215,32 @@ func (w *writer) encodeWithLock(
 				PoliciesList: pl,
 			}
 			err = encoder.EncodeBatchTimerWithPoliciesList(btp)
+			if err != nil {
+				break
+			}
+
+			sizeAfter := buffer.Len()
+			if sizeAfter > w.flushSize {
+				oldBufferedEncoder := w.prepareEnqueue(encoder, sizeBefore)
+
+				// Update variables in case they are used in a future iteration of the loop.
+				bufferedEncoder = encoder.Encoder()
+				buffer = bufferedEncoder.Buffer()
+				sizeBefore = buffer.Len()
+
+				// Unlock the encoder before we enqueue the old buffer to ensure other
+				// goroutines have an oppurtunity to encode metrics while larger timer
+				// batches are being encoded.
+				encoder.Unlock()
+
+				err = w.queue.Enqueue(oldBufferedEncoder)
```

Comment:
nit: move the enqueue logic to `enqueueBuffer` so you don't need to repeat this below.